### PR TITLE
`Programming exercises`: Close the diff modal when leaving the page

### LINKED ISSUE #9351: Back Gesture in Diffview
### Describe the bug

When using a back gesture like on mac swiping with two fingers, the diffview does not get close and breaks.

### To Reproduce

1. Go as an instructor to the details view of an programming exercise
2. click on review changes to open the diff view
3. back a back gesture like swiping with two fingers on mac

### Expected behavior

the diff view should close

### Screenshots

https://github.com/user-attachments/assets/8d2b8c33-6d40-4b02-b449-67144434fe2e

### Which version of Artemis are you seeing the problem on?

7.5.3

### What browsers are you seeing the problem on?

Other (specify in "Additional context")

### Additional context

Arc Browser: Chromium Based

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by pzdr7**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

#### Changes affecting Programming Exercises
- [x] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).

### Motivation and Context

Closes #9351

The modal for the git diff view did not close when the user left the page. Furthermore, its size decreased, leading to some visual issues.

### Description

Closes the modal and moves the windowClass to the global style declarations.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming exercise

1. Navigate to Course Management > Your course > Exercises > Your programming exercises (click on its name)
2. Scroll down and click "Review changes".
3. Without closing the modal, click the back button in your browser.
4. Verify that the modal closes when you leave the page.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

Unchanged

### Screenshots

No UI changes

## Release Notes

- **New Features**
	- Enhanced modal management in the Programming Diff Report Detail component for improved resource handling.

- **Style Changes**
	- Simplified styling rules for the detail overview list and modal dialogs.
	- Updated modal class names and adjusted maximum widths for better consistency.
	- Removed styling for the Ace editor Dracula theme to streamline appearance.

- **Bug Fixes**
	- Removed outdated hover state styling for form controls to improve user experience.

### CODE REVIEW

File: src/main/webapp/content/scss/global.scss [hunk 1]
```diff
@@ -827,14 +826,12 @@ Course Info Bar
     }
 }
 
-.warning-modal-window .modal-dialog {
-    width: 50%;
+.diff-view-modal .modal-dialog {
+    max-width: 80vw;
 }
```
Comment (pzdr7) [inline src/main/webapp/content/scss/global.scss:831]: I had to move this declaration here to ensure that we wouldn't encounter visual glitches when leaving the page

File: src/main/webapp/content/scss/global.scss [hunk 2]
```diff
@@ -827,14 +826,12 @@ Course Info Bar
...
 
-/* additional style to overwrite form-control style from parent components */
-.form-control.ace_editor {
-    font-size: 12px;
-    line-height: normal;
```
Comment (pzdr7) [inline src/main/webapp/content/scss/global.scss:837]: Ace is no longer in Artemis, so I removed this

File: src/main/webapp/app/detail-overview-list/components/programming-diff-report-detail/programming-diff-report-detail.component.ts [hunk 3]
```diff
@@ -25,15 +25,20 @@ export class ProgrammingDiffReportDetailComponent {
...
+
     showGitDiff(gitDiff?: ProgrammingExerciseGitDiffReport) {
         if (!gitDiff) {
             return;
         }
 
-        const modalRef = this.modalService.open(GitDiffReportModalComponent, { windowClass: 'diff-view-modal' });
-        modalRef.componentInstance.report = gitDiff;
+        this.modalRef = this.modalService.open(GitDiffReportModalComponent, { windowClass: 'diff-view-modal' });
+        this.modalRef.componentInstance.report = gitDiff;
```
[the remark this answers is not in the document]
Comment (pzdr7) [reply src/main/webapp/app/detail-overview-list/components/programming-diff-report-detail/programming-diff-report-detail.component.ts:42]: I moved this

**Review by SimonEntholzer — DISMISSED** [review]
Works on Ts2

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by magaupp — APPROVED** [review]
Tested on TS5. Back navigation closes modal.

**Review by EneaGore — APPROVED** [review]
Tested on TS5, works as expected.

**Review by edkaya — APPROVED** [review]
Tested on TS5, works as expected

**Review by krusche — APPROVED** [review]

**Pull request merged**